go/private/rules/transition.bzl

```diff
@@ -366,6 +398,39 @@ go_transition.
 """,
 )
 
+def _non_go_transition_impl(settings, attr):
+    """Sets all Go settings to the values they had before the last go_transition.
+
+    non_go_transition sets all of the //go/config settings to the value they had
+    before the last go_transition. This should be used on all attributes of
+    go_library/go_binary/go_test that are built in the target configuration and
+    do not constitute advertise any Go providers.
+
+    Examples: This transition is applied to the 'data' attribute of go_binary so
+    that other Go binaries used at runtime aren't affected by a non-standard
+    link mode set on the go_binary target, but still use the same top-level
+    settings such as e.g. race instrumentation.
+    """
+    new_settings = {}
+    for key in TRANSITIONED_GO_SETTING_KEYS:
+        original_key = _original_setting_key(key)
+        original_value = settings[original_key]
+        if original_value:
+            # Reset to the original value and clear it.
+            new_settings[key] = json.decode(original_value)
+            new_settings[original_key] = ""
+        else:
+            new_settings[key] = settings[key]
+            new_settings[original_key] = settings[original_key]
+
+    return new_settings
+
+non_go_transition = transition(
+    implementation = _non_go_transition_impl,
+    inputs = TRANSITIONED_GO_SETTING_KEYS + _ORIGINAL_SETTING_KEYS,
+    outputs = TRANSITIONED_GO_SETTING_KEYS + _ORIGINAL_SETTING_KEYS,
```

Comment:
Do `outputs` need to include `_ORIGINAL_SETTING_KEYS`? After resetting to the original values, do we still need the values for `_ORIGINAL_SETTING_KEYS`?